pyrft/other.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+"""
+ARI semi modified function files
+"""
+import numpy as np
+from scipy.ndimage import label
+from scipy.stats import norm
+
+from nilearn.input_data import NiftiMasker
+from nilearn.image import get_data, math_img
+
+def ari(pvalues, subset, alpha= 0.1 ):
+    """ ari implements all resolution inference (Rosenblatt 2018, see below) and 
+    calulates bounds on the TDP for p-values within the specified subset
+
+    Parameters
+    ----------
+    pvalues:  a numpy.nd array,
+        containing the p-values
+
+    subset:   numpy boolean array,
+        the same size as pvalues indicating which elements
+        to use in the subset or 
+
+    alpha: float or list, optional
+        level of control on the true positive rate, aka true dsicovery
+        proportion
+
+    Returns
+    -------
+    tdp:  float,
+        the true discovery proportion found by applying ARI
+
+    Note
+    ----
+    This implements the method described in:
+
+    Rosenblatt JD, Finos L, Weeda WD, Solari A, Goeman JJ. All-Resolutions
+    Inference for brain imaging. Neuroimage. 2018 Nov 1;181:786-796. doi:
+    10.1016/j.neuroimage.2018.07.060
+    """
+
+    hommel_value = _compute_hommel_value(pvalues, alpha)
+    subset_pvalues = pvalues[subset]
+    tdp = _true_positive_fraction(subset_pvalues, hommel_value, alpha)
+    
+    return tdp
+
+def _compute_hommel_value(pvalues, alpha = 0.1):
+    """ This function computes the hommel value needed for all resolution 
+    inference
+
+    Parameters
+    ---------------------

```

Comment:
```suggestion
    ----------
``` As mny dash as there are characters in 'Parameters'

Reply:
fixed